test/Dialect/FIRRTL/annotations-errors.mlir

```diff
@@ -239,3 +239,19 @@ firrtl.circuit "GCTDataTapUnsupportedLiteral" attributes {rawAnnotations = [{
     %tap = firrtl.wire : !firrtl.uint<1>
   }
 }
+
+// -----
+// Reject annotations on references.
+
+// expected-error @below {{cannot find name 'out' in RefAnno}}
+// expected-error @below {{Unable to resolve target of annotation}}
```

Comment:
Is there a way to improve the error reporting here to be less generic? I.e., how can the error message also include information about that references are not allowed (like with the instance port case)?

Reply:
Totally, thanks for the QC.  Misunderstood how the code used this + cache.  Now cache will contain all targets, resolution code can reject as it wants.  Will push updated version soon, thanks, this bugged me too :+1:.  As a future extension we could extend AnnoTarget to provide an error location `Location getLoc()` which may be useful elsewhere too (?), but for now reporting on the circuit.